geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementListOperationsResult.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.api;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.management.runtime.OperationResult;
+
+/**
+ * returned from {@link ClusterManagementService#list(ClusterManagementOperation)}
+ *
+ * @param <V> the result type associated with the operation type being listed
+ */
+@Experimental
+public class ClusterManagementListOperationsResult<V extends OperationResult>
```

Comment:
Could this class be made an interface and have an internal class that implements the interface? It would be nice for "setResult" and the constructors to not be an external api.

Note that some of the methods on the class are missing javadocs which is okay if it was internal but every method in external should hav e javadocs

Reply:
made it immutable, that's about as good as we can get for now and still be JSON-deserializable (without adding a `class` attribute exposing the implementation class in the JSON)

Reply:
added missing javadocs